docs/build.md

```diff
@@ -17,12 +17,15 @@ commands provided below.
 - [Protocol Buffer Compiler](https://grpc.io/docs/protoc-installation/)
 - [buf](https://docs.buf.build/installation)
 - [musl libc](https://musl.libc.org)
+- [BPF Linker](https://github.com/aya-rs/bpf-linker)
+
 
 ##### Ubuntu
 
 ```bash
 sudo apt-get install -y protobuf-compiler; # Protocol Buffer Compiler
 sudo apt-get install -y musl-tools; # musl libc
+sudo apt-get install -y build-essentials; # gcc compiler
```

Comment:
I have concerns about licensing issues and want to make sure we get a review from @krisnova to see if this is ok to add before we merge anything.

Reply:
not trying to resolve as that's up to @krisnova, but the use of GPL for eBPF is a well-known thing and other projects that embed eBPF programs end up dual-licensed which is _fine_ but a bit of a pain.

can we just pass in a license?

Reply:
The licensing pertains to BPF helper functions in the kernel, [some of them are GPL-ed and some are not](https://github.com/iovisor/bcc/blob/master/docs/kernel-versions.md#helpers). This PR is using [`BPF_FUNC_perf_event_output()`](https://github.com/JeroenSoeters/aurae/blob/posix-signals-api/aurae-ebpf/src/main.rs#L44) which has a GPL license. I think we would need to dual license only the BPF code [like Cilium does](https://github.com/cilium/cilium#license).

Reply:
If the function we are using is GPL-ed and everything was working (which I assume is true given the discord post), then **`aya` must be assuming we are licensed under GPL**.

Some food-for-thought:
- If some code is going to be GPL-ed, should we put it in a separate repo with its own builds and releases? Even if it doesn't make a technical difference, I think that is a better approach then what I assume is the alternative -- update the license files to say "this is the default unless otherwise indicated".
- Do we need to update the CLA?

Also, I want to clarify that I'm not trying to suggest the direction we go in; I don't have the knowledge needed to make those suggestions.

Reply:
> aya must be assuming we are licensed under GPL.

It's not just `aya`, you literally have to tell the kernel that the BPF program you're loading is GPL licensed. This is what that `syscall` looks like:

```bpf(BPF_PROG_LOAD, {prog_type=BPF_PROG_TYPE_TRACEPOINT, insn_cnt=31, insns=0x5555593ae5c0, license="GPL", log_level=0, log_size=0, log_buf=NULL, kern_version=KERNEL_VERSION(6, 1, 8), prog_flags=0, prog_name="signals", prog_ifindex=0, expected_attach_type=BPF_CGROUP_INET_INGRESS, prog_btf_fd=0, func_info_rec_size=0, func_info=NULL, func_info_cnt=0, line_info_rec_size=0, line_info=NULL, line_info_cnt=0, attach_btf_id=0, attach_prog_fd=0, fd_array=NULL}, 144)```

I'm not sure if we would want to create a separate repo for the BPF code, though. We'd have to run a private crate registry, and we'd be introducing a lot of friction in the development process, not to mention that for every change, we would now need 2 PRs instead of 1.

If you look at some of the other popular OSS eBPF projects like Cilium and Falco, they typically have a mention somewhere of which files are dual licensed and have a different license header for those files.

https://github.com/cilium/cilium#license
https://github.com/falcosecurity/libs/blob/master/NOTICES

Reply:
dual-licensing may not be an issue for our use-case as we're not shipping this as a library to be integrated into other projects.  ie, anyone _using_ aurae is not going to be subject to the terms in GPL unless they're building code linking aurae in, which is not what i'd expect.

but IANAL and i agree @krisnova should have final say.  also, i don't feel qualified to assess the eBPF parts of this PR :)